common/client/head_poller_test.go

```diff
@@ -0,0 +1,78 @@
+package client
+
+import (
+	"errors"
+	"math/big"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func Test_Poller(t *testing.T) {
```

Comment:
It would be nice to increase test coverage and add few corner cases. Could you add a test cases that ensures `Unsubscribe` exits even if no one is listening to the `p.channel`? And one that ensure we can call `Unsubscribe` twice without panic?

Reply:
Yeah for sure, I plan on increasing test coverage.